batch/plugins/org.jboss.tools.batch.core/src/org/jboss/tools/batch/internal/core/impl/PreferredPackageManager.java

```diff
@@ -0,0 +1,76 @@
+/******************************************************************************* 
+ * Copyright (c) 2015 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/ 
+package org.jboss.tools.batch.internal.core.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.core.runtime.QualifiedName;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.IPackageFragment;
+import org.jboss.tools.batch.core.BatchArtifactType;
+import org.jboss.tools.batch.core.BatchCorePlugin;
+import org.jboss.tools.batch.core.IBatchArtifact;
+import org.jboss.tools.batch.core.IBatchProject;
+import org.jboss.tools.common.EclipseUtil;
+
+public class PreferredPackageManager {
+	private static final String QUALIFIED_NAME_PREFIX = "create.artifact.package.";
+	
+	public static IPackageFragment getPackageSuggestion(IBatchProject batchProject, BatchArtifactType type) {
+		ArrayList<BatchArtifactType> list = new ArrayList<BatchArtifactType>();
+		list.add(type);
+		return getPackageSuggestion(batchProject, list);
+	}
+	
+	public static IPackageFragment getPackageSuggestion(IBatchProject batchProject, List<BatchArtifactType> types) {
+		IProject project = batchProject.getProject();
+		QualifiedName qualifiedName = new QualifiedName("", QUALIFIED_NAME_PREFIX + types.get(0).toString());
```

Comment:
This will throw an exception if types is empty. What's the point of passing the list of types if the method is using the first  item only? If we don't support list of types then we don't need this method. getPackageSuggestion(IBatchProject batchProject, BatchArtifactType type) is enough.